### PR TITLE
Exclude dimensions from star-tree index stored type check

### PR DESCRIPTION
**Pull request by Jackie-Jiang**
Related to #12164 and #12554

Do not check stored type for dimensions to avoid mismatching columns with '__' in column name

### CODE REVIEW

File: pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/store/StarTreeIndexReader.java [hunk 1]
```diff
@@ -67,17 +66,11 @@ public class StarTreeIndexReader implements Closeable {
    */
   public StarTreeIndexReader(File segmentDirectory, SegmentMetadataImpl segmentMetadata, ReadMode readMode)
       throws IOException, ConfigurationException {
-    Preconditions.checkNotNull(segmentDirectory);
```
Comment (jadami10) [inline pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/store/StarTreeIndexReader.java:70]: what's the reason for removing all these? It's guaranteed elsewhere?

File: pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/store/SegmentLocalFSDirectory.java [hunk 2]
```diff
@@ -268,7 +267,7 @@ private synchronized void loadData()
       default:
         break;
     }
-    if (CollectionUtils.isNotEmpty(_segmentMetadata.getStarTreeV2MetadataList())) {
+    if (_segmentMetadata.getStarTreeV2MetadataList() != null) {
```
Comment (jadami10) [inline pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/store/SegmentLocalFSDirectory.java:270]: looks like you've updated it to just handle an empty list instead?

File: pinot-segment-local/src/test/java/org/apache/pinot/segment/local/startree/v2/builder/StarTreeIndexSeparatorTest.java [hunk 3]
```diff
@@ -18,85 +18,63 @@
...
-  private PropertiesConfiguration _metadataProperties;
-  private final StarTreeV2BuilderConfig _builderConfig = StarTreeV2BuilderConfig.fromIndexConfig(
-      new StarTreeIndexConfig(
-          Lists.newArrayList("AirlineID", "Origin", "Dest"),
-          Lists.newArrayList(),
-          Lists.newArrayList("count__*", "max__ArrDelay"),
-          null,
-          10));
+  private final StarTreeV2BuilderConfig BUILDER_CONFIG = StarTreeV2BuilderConfig.fromIndexConfig(
+      new StarTreeIndexConfig(List.of("AirlineID", "Origin", "Dest"), List.of(), List.of("count__*", "max__ArrDelay"),
```
Comment (jadami10) [inline pinot-segment-local/src/test/java/org/apache/pinot/segment/local/startree/v2/builder/StarTreeIndexSeparatorTest.java:45]: should one of the dimensions specifically have the `<agg>__<name>` format?

**Review by jadami10 — COMMENTED** [review]
i'm fairly new to this code. Thanks for jumping on this!

File: pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/store/StarTreeIndexReader.java (hunk 1 above)
Comment (Jackie-Jiang) [reply pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/store/StarTreeIndexReader.java:70]: Yes. It is internal class, thus we can control all the caller behavior. We usually annotate argument that can be null with `@Nullable`, and skip the null check for performance reason

File: pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/store/SegmentLocalFSDirectory.java (hunk 2 above)
Comment (Jackie-Jiang) [reply pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/store/SegmentLocalFSDirectory.java:270]: It cannot be empty, only null or non-empty. I modified this place to make it consistent with other callers

File: pinot-segment-local/src/test/java/org/apache/pinot/segment/local/startree/v2/builder/StarTreeIndexSeparatorTest.java (hunk 3 above)
Comment (Jackie-Jiang) [reply pinot-segment-local/src/test/java/org/apache/pinot/segment/local/startree/v2/builder/StarTreeIndexSeparatorTest.java:45]: Good suggestion. Updated `BaseStarTreeV2Test` to capture that

**Pull request merged**